airbyte-ci/connectors/connectors_qa/src/connectors_qa/checks/version.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+
+from typing import Any, Dict
+
+import requests  # type: ignore
+import semver  # type: ignore
+import yaml  # type: ignore
+from connector_ops.utils import Connector  # type: ignore
+
+from connectors_qa import consts
+from connectors_qa.models import Check, CheckCategory, CheckResult
+
+
+class CheckVersionIncrement(Check):
+    """Check that the connector version was incremented if files were modified."""
+
+    category = CheckCategory.VERSION
+    name = "Connector Version Increment Check"
+    description = "Validates that the connector version was incremented if files were modified."
+
+    # _BYPASS_CHECK_FOR = [
+    #     consts.METADATA_FILE_NAME,
+    #     "acceptance-test-config.yml",
+    #     "README.md",
+    #     "bootstrap.md",
+    #     ".dockerignore",
+    #     "unit_tests",
+    #     "integration_tests",
+    #     "src/test",
+    #     "src/test-integration",
+    #     "src/test-performance",
+    #     "build.gradle",
+    #     "erd",
+    #     "build_customization.py",
+    # ]
+
+    def _get_master_metadata(self, connector: Connector) -> Dict[str, Any] | None:
+        """Get the metadata from the master branch or None if unable to retrieve."""
+        github_url_prefix = "https://raw.githubusercontent.com/airbytehq/airbyte/master/airbyte-integrations/connectors"
+        master_metadata_url = f"{github_url_prefix}/{connector.technical_name}/{consts.METADATA_FILE_NAME}"
```

Comment:
I don't know if https fetch will will work when we later want to run also on the `airbyte-enterprise` repo, which is private.

Might be worth porting as a git-based operation, in which case I believe permissions would be handled internally to git.